src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -710,6 +747,85 @@ private String upper(String property, String culture) {
         return new Tuple<>(name, def(name, "(int) " + getPropertyOrStringValue(field) + ".charAt(0)"));
     }
 
+    private Tuple<String, String> ifFunc(List<KVValue> paramers) {
+        String expr1 = paramers.get(1).value.toString();
+        String expr2 = paramers.get(2).value.toString();
+
+        if (paramers.get(0).value instanceof MethodField) {
+            String condition = getScriptText((MethodField) paramers.get(0).value);
+            return ifFunc(condition, expr1, expr2);
+        } else if (paramers.get(0).value instanceof SQLBooleanExpr) {
+            Boolean condition = ((SQLBooleanExpr) paramers.get(0).value).getValue();
```

Comment:
could you give an example of query that will trigger that condition? I'm trying to understand how it end's up as true/false that we analyze in the method at line # 783

Reply:
@galkk Yes sure. 
When we give a query like this: `SELECT IF(true, 1, 0)`. The first param (KVValue) would get a KVValue with `key = null` and `value = (SQLBooleanExpr) "xTRUE"`, and the value of this boolean expr is true.